storage/src/iter.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+//! Internal utilities for working with iterators.
+
+/// Writes an iterator of displayable items to a writer, separated by a specified separator.
+///
+/// - If `limit` is `Some(0)`, it will only write the number of hidden items.
+/// - If `limit` is `Some(n)`, it will write at most `n` items, followed by a message
+///   indicating how many more items were not written.
+/// - If `limit` is `None`, it will write all items without any limit.
+///
```

Comment:
Add a caution about this function not working with iterators that do not terminate like `once(1).cycle()`